src/test/java/seedu/address/ui/MainAppTest.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.ui;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.testfx.api.FxAssert.verifyThat;
+import static org.testfx.util.NodeQueryUtils.isVisible;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testfx.api.FxToolkit;
+import org.testfx.framework.junit5.ApplicationTest;
+
+import javafx.scene.control.Label;
+import javafx.scene.input.KeyCode;
+import seedu.address.MainApp;
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.ClearCommand;
+import seedu.address.testutil.PersonBuilder;
+
+public class MainAppTest extends ApplicationTest {
+
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        // Use FxToolkit to launch the JavaFX Application correctly
+        FxToolkit.registerPrimaryStage();
+        FxToolkit.setupApplication(MainApp::new);
+    }
+
+    @BeforeEach
+    public void clearEverything() throws Exception {
+        clickOn("#commandTextField");
+        write(ClearCommand.COMMAND_WORD);
+        push(KeyCode.ENTER);
+    }
+
+    @Test
+    public void personHasAllDetailsShown() {
+        //adds a person
+        String name = PersonBuilder.DEFAULT_NAME;
+        String phone = PersonBuilder.DEFAULT_PHONE;
+        String email = PersonBuilder.DEFAULT_EMAIL;
+        String address = PersonBuilder.DEFAULT_ADDRESS;
+        String schedule = PersonBuilder.DEFAULT_SCHEDULE;
+        String subject = PersonBuilder.DEFAULT_SUBJECT;
+        String fee = PersonBuilder.DEFAULT_FEE;
+        String paid = PersonBuilder.DEFAULT_PAID;
+        String owed = PersonBuilder.DEFAULT_OWED_AMOUNT;
+
+        //Add a person to the list
+        clickOn("#commandTextField");
+
+        write(AddCommand.COMMAND_WORD);
+        write(" n/" + name);
+        write(" p/" + phone);
+        write(" e/" + email);
+        write(" a/" + address);
+        write(" time/" + schedule);
+        write(" subject/" + subject);
+        write(" fee/" + fee);
+        write(" paid/" + paid);
+        write(" owed/" + owed);
```

Comment:
this might be bug-prone and violates DRY principle. Will it be better to use e.g. write(" " + PREFIX_NAME + name) instead?